### ISSUE TITLE
When running function fit_ssm : Error in MakeADFun. Is someone familiar?

### ISSUE THREAD

**Issue by TamaraVallina** [ianjonsen/aniMotum#46]
Hi!

I commented on issue 25  (closed) yesterday, but I figured it might be better to post this in a seperate issue. I'm running into the error below when running 'fit_ssm'; does somebody have an idea of whats going wrong?
```
Error in MakeADFun(data, parameters, map = map, random = rnd, hessian = TRUE,  : 
  Only numeric matrices, vectors, arrays, factors, lists or length-1-characters can be interfaced
```
I have run this script many times before, and it has always worked for me. Last week, I updated R to the latest version, after which this error occurred. I made sure to reinstall TMB and glmmTMB packages from source. I have tried reinstalling AniMotum, but the error stays the same. Some information on the session:
```

> sessionInfo()
R version 4.3.2 (2023-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8    LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                           LC_TIME=English_United States.utf8    

time zone: Europe/Amsterdam
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] glmmTMB_1.1.8       TMB_1.9.6           Matrix_1.6-2        ggspatial_1.1.9     ggplot2_3.4.4       stars_0.6-4         abind_1.4-5        
 [8] sf_1.0-14           adehabitatLT_0.3.27 CircStats_0.2-6     boot_1.3-28.1       MASS_7.3-60         adehabitatMA_0.3.16 ade4_1.7-22        
[15] sp_2.1-1            aniMotum_1.2        momentuHMM_1.5.5    lubridate_1.9.3     mapview_2.11.2      lattice_0.22-5      crawl_2.3.0        
[22] vtable_1.4.6        kableExtra_1.3.4    fuzzyjoin_0.1.6     tidyr_1.3.0         dplyr_1.1.3         data.table_1.14.8  

loaded via a namespace (and not attached):
 [1] DBI_1.1.3           sandwich_3.0-2      rlang_1.1.2         magrittr_2.0.3      e1071_1.7-13        compiler_4.3.2      mgcv_1.9-0         
 [8] png_0.1-8           systemfonts_1.0.5   vctrs_0.6.4         rvest_1.0.3         stringr_1.5.0       pkgconfig_2.0.3     fastmap_1.1.1      
[15] leafem_0.2.3        utf8_1.2.4          rmarkdown_2.25      nloptr_2.0.3        purrr_1.0.2         xfun_0.41           satellite_1.0.4    
[22] jsonlite_1.8.7      gmm_1.8             terra_1.7-55        parallel_4.3.2      R6_2.5.1            stringi_1.7.12      numDeriv_2016.8-1.1
[29] Rcpp_1.0.11         iterators_1.0.14    knitr_1.45          zoo_1.8-12          base64enc_0.1-3     splines_4.3.2       timechange_0.2.0   
[36] tidyselect_1.2.0    rnaturalearth_0.3.4 rstudioapi_0.15.0   doParallel_1.0.17   codetools_0.2-19    doRNG_1.8.6         tibble_3.2.1       
[43] withr_2.5.2         evaluate_0.23       tmvtnorm_1.5        units_0.8-4         proxy_0.4-27        xml2_1.3.5          pillar_1.9.0       
[50] rngtools_1.5.2      KernSmooth_2.23-22  traipse_0.3.0       foreach_1.5.2       stats4_4.3.2        generics_0.1.3      munsell_0.5.0      
[57] scales_1.2.1        minqa_1.2.6         class_7.3-22        glue_1.6.2          tools_4.3.2         lme4_1.1-35.1       webshot_0.5.5      
[64] mvtnorm_1.2-3       grid_4.3.2          crosstalk_1.2.0     colorspace_2.1-0    nlme_3.1-163        patchwork_1.1.3     raster_3.6-26      
[71] cli_3.6.1           fansi_1.0.5         viridisLite_0.4.2   svglite_2.1.2       Brobdingnag_1.2-9   gtable_0.3.4        digest_0.6.33      
[78] classInt_0.4-10     htmlwidgets_1.6.2   htmltools_0.5.7     lifecycle_1.0.4     leaflet_2.2.0       httr_1.4.7 

```
And some more info on the script I ran:
```

#---------------------------
# fitting SSM 
#---------------------------
str(data)

# 'data.frame':	241 obs. of  7 variables:
# $ id    : chr  "8-1" "8-1" "8-1" "8-1" ...
# $ date  : POSIXct, format: "2018-10-19 18:37:01" "2018-10-19 18:37:43" "2018-10-19 18:37:57" "2018-10-19 18:38:11" ...
# $ lon   : num  4.7 4.7 4.7 4.7 4.7 ...
# $ lat   : num  52.8 52.8 52.8 52.8 52.8 ...
# $ lonerr: num  0.01034 0.00773 0.00758 0.00783 0.00801 ...
# $ laterr: num  0.00626 0.00468 0.00458 0.00474 0.00484 ...
# $ lc    : chr  "GL" "GL" "GL" "GL" ...

ssm_ID <- fit_ssm(x=data, 
                  model="crw",
                  vmax=20, # m/s
                  time.step = as.numeric(1/60), # wants input in hours, I want one minute
                  spdf = FALSE)

# fitting crw SSM to 1 tracks...
# Problem with these data entries:
#   GLerr 
# 16 
# Error in MakeADFun(data, parameters, map = map, random = rnd, hessian = TRUE,  : 
#                      Only numeric matrices, vectors, arrays, factors, lists or length-1-characters can be interfaced

```
The first part of the error, 'problem with these data entries' is consistent for my tries in with different datasets I try (I run these separately for diff. animals), where these datasets worked fine before. Also, the required structure of the dataset seems to be correct. I don't think the true problem lies here. I don't know what to try to solve the 'MakeADFun' error. Does anyone have any ideas?

**Comment by ianjonsen** [comment]
If you change the variable names `lonerr` and `laterr` to `x.sd` and `y.sd`, respectively, then your code will run without that error. In `aniMotum 1.2`, the error SD variable names have been changed from `lonerr` and `laterr` to `x.sd` and `y.sd` to accommodate a variety of tracking data types in the Generic Location (`GL`) class beyond just light-level geolocations. See `?fit_ssm` for more details regarding the `Generic Location` (GL) class.

I've added a check for `lonerr` and `laterr` variable names. If present, the `format_data()` and `fit_ssm()` now return a warning and change the names automatically before proceeding.
